In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import warnings 
warnings.filterwarnings('ignore')

In [2]:
music_sales_df = pd.read_csv('../data/Usable_Data/Full_Music_Sales_Data_Clean.csv')

## Creating a map 

In this workbook, I created a map to categorize all of my formats into three categories which I named "Modes": Analog, digital, and streaming. I added an additional column to my df to be able to carry out analyses based on Mode.

In [3]:
music_sales_df

Format            Metric  Year  Format_Value_Num_Billions  \
0               CD             Units  1973                        NaN   
1               CD             Units  1974                        NaN   
2               CD             Units  1975                        NaN   
3               CD             Units  1976                        NaN   
4               CD             Units  1977                        NaN   
...            ...               ...   ...                        ...   
2956  Vinyl Single  Value (Adjusted)  2015                        NaN   
2957  Vinyl Single  Value (Adjusted)  2016                        NaN   
2958  Vinyl Single  Value (Adjusted)  2017                        NaN   
2959  Vinyl Single  Value (Adjusted)  2018                        NaN   
2960  Vinyl Single  Value (Adjusted)  2019                        NaN   

      Format_Value_Num_Millions  Actual_Value  
0                           NaN           NaN  
1                           NaN           NaN  
2                           NaN           NaN  
3                           NaN           NaN  
4                           NaN           NaN  
...                         ...           ...  
2956                        6.2      6.205390  
2957                        5.2      5.198931  
2958                        6.3      6.339678  
2959                        5.4      5.386197  
2960                        6.8      6.795946  

[2961 rows x 6 columns]

In [4]:
music_sales_df['Format'].unique()

array(['CD', 'CD Single', 'Cassette', 'Cassette Single', 'LP/EP',
       'Vinyl Single', '8 - Track', 'Other Tapes',
       'Music Video (Physical)', 'DVD Audio', 'SACD', 'Download Single',
       'Download Album', 'Kiosk', 'Download Music Video',
       'Ringtones & Ringbacks', 'Paid Subscriptions',
       'Limited Tier Paid Subscription',
       'On-Demand Streaming (Ad-Supported)',
       'Other Ad-Supported Streaming', 'Other Digital',
       'Paid Subscription', 'SoundExchange Distributions',
       'Synchronization'], dtype=object)

In [5]:
format_map = {'CD':'Digital',
              'CD Single':'Digital',
              'Cassette':'Analog',
              'Cassette Single':'Analog',
              'LP/EP':'Analog',
       'Vinyl Single':'Analog',
              '8 - Track':'Analog',
              'Other Tapes':'Analog',
       'Music Video (Physical)':'Digital',
              'DVD Audio':'Digital',
              'SACD':'Digital',
              'Download Single':'Digital',
       'Download Album':'Digital',
              'Kiosk':'Digital',
              'Download Music Video':'Digital',
       'Ringtones & Ringbacks':'Digital',
              'Paid Subscriptions':'Streaming',
       'Limited Tier Paid Subscription':'Streaming',
       'On-Demand Streaming (Ad-Supported)':'Streaming',
       'Other Ad-Supported Streaming':'Streaming',
              'Other Digital':'Digital',
       'Paid Subscription':'Streaming',
              'SoundExchange Distributions':'Digital',
       'Synchronization':'Digital'}

In [6]:
music_sales_df['Mode'] = music_sales_df['Format'].map(format_map)

In [7]:
music_sales_df

Format            Metric  Year  Format_Value_Num_Billions  \
0               CD             Units  1973                        NaN   
1               CD             Units  1974                        NaN   
2               CD             Units  1975                        NaN   
3               CD             Units  1976                        NaN   
4               CD             Units  1977                        NaN   
...            ...               ...   ...                        ...   
2956  Vinyl Single  Value (Adjusted)  2015                        NaN   
2957  Vinyl Single  Value (Adjusted)  2016                        NaN   
2958  Vinyl Single  Value (Adjusted)  2017                        NaN   
2959  Vinyl Single  Value (Adjusted)  2018                        NaN   
2960  Vinyl Single  Value (Adjusted)  2019                        NaN   

      Format_Value_Num_Millions  Actual_Value     Mode  
0                           NaN           NaN  Digital  
1                           NaN           NaN  Digital  
2                           NaN           NaN  Digital  
3                           NaN           NaN  Digital  
4                           NaN           NaN  Digital  
...                         ...           ...      ...  
2956                        6.2      6.205390   Analog  
2957                        5.2      5.198931   Analog  
2958                        6.3      6.339678   Analog  
2959                        5.4      5.386197   Analog  
2960                        6.8      6.795946   Analog  

[2961 rows x 7 columns]

#### I created a map for the different formats. Then I added a column that indicates whether a format falls into Digital, Analog, or Streaming

In [8]:
cols_to_keep = ['Format','Metric','Year','Actual_Value','Mode']

In [9]:
music_sales_df2 = music_sales_df[cols_to_keep]

In [10]:
music_sales_df2

Format            Metric  Year  Actual_Value     Mode
0               CD             Units  1973           NaN  Digital
1               CD             Units  1974           NaN  Digital
2               CD             Units  1975           NaN  Digital
3               CD             Units  1976           NaN  Digital
4               CD             Units  1977           NaN  Digital
...            ...               ...   ...           ...      ...
2956  Vinyl Single  Value (Adjusted)  2015      6.205390   Analog
2957  Vinyl Single  Value (Adjusted)  2016      5.198931   Analog
2958  Vinyl Single  Value (Adjusted)  2017      6.339678   Analog
2959  Vinyl Single  Value (Adjusted)  2018      5.386197   Analog
2960  Vinyl Single  Value (Adjusted)  2019      6.795946   Analog

[2961 rows x 5 columns]

#### Since the data frame already contains so much data (all the sales, revenue, and revenue adjusted for inflation), I decided to just simplify it to 5 columns. 

In [11]:
music_sales_df2['Metric'].unique()

array(['Units', 'Value', 'Value (Adjusted)'], dtype=object)

#### Units refers to sales in terms of volume. Both Value and Value (Adjusted) refer to sales in revenue, however one is adjusted for inflation. I will use adjusted for inflation as I am interested in seeing the impact of different formats on music sales and taking inflation into consideration can give a fuller picture and make years more comparable. I will remove rows with 'Value' in the Metric column.

In [12]:
music_sales_df2.drop(music_sales_df2.loc[music_sales_df2['Metric']=='Value'].index, inplace=True)

In [13]:
music_sales_df2['Metric'].unique()

array(['Units', 'Value (Adjusted)'], dtype=object)

In [14]:
music_sales_df2.sample(20)

Format            Metric  Year  Actual_Value  \
613                         Kiosk             Units  1975           NaN   
2885              Synchronization  Value (Adjusted)  1991           NaN   
1935                           CD  Value (Adjusted)  1981           NaN   
602                Download Album             Units  2011    103.900000   
421        Music Video (Physical)             Units  2018      1.395058   
2821  SoundExchange Distributions  Value (Adjusted)  1974           NaN   
2597                Other Digital  Value (Adjusted)  1985           NaN   
2014                    CD Single  Value (Adjusted)  2013      2.684057   
2901              Synchronization  Value (Adjusted)  2007           NaN   
2911              Synchronization  Value (Adjusted)  2017    242.063251   
1881                    8 - Track  Value (Adjusted)  1974   2848.008609   
2817                         SACD  Value (Adjusted)  2017      0.887142   
767            Paid Subscriptions             Units  1988           NaN   
342                   Other Tapes             Units  1986      0.000000   
521               Download Single             Units  1977           NaN   
20                             CD             Units  1993    495.400000   
2129                    DVD Audio  Value (Adjusted)  1987           NaN   
2464       Music Video (Physical)  Value (Adjusted)  1993    377.381579   
2344                        Kiosk  Value (Adjusted)  2014      2.789052   
2937                 Vinyl Single  Value (Adjusted)  1996     77.397753   

           Mode  
613     Digital  
2885    Digital  
1935    Digital  
602     Digital  
421     Digital  
2821    Digital  
2597    Digital  
2014    Digital  
2901    Digital  
2911    Digital  
1881     Analog  
2817    Digital  
767   Streaming  
342      Analog  
521     Digital  
20      Digital  
2129    Digital  
2464    Digital  
2344    Digital  
2937     Analog

#### I will also get rid of rows with NaN, as this indicates that a certain format was not available during a specific year. It's not really useful to have the empty rows, it just creates more confusion and more unnecessary rows.

In [15]:
music_sales_df.shape

(2961, 7)

In [16]:
music_sales_df2 = music_sales_df2.dropna().reset_index(drop=True)

In [17]:
music_sales_df2

Format            Metric  Year  Actual_Value     Mode
0              CD             Units  1983      0.800000  Digital
1              CD             Units  1984      5.800000  Digital
2              CD             Units  1985     22.600000  Digital
3              CD             Units  1986     53.000000  Digital
4              CD             Units  1987    102.100000  Digital
..            ...               ...   ...           ...      ...
895  Vinyl Single  Value (Adjusted)  2015      6.205390   Analog
896  Vinyl Single  Value (Adjusted)  2016      5.198931   Analog
897  Vinyl Single  Value (Adjusted)  2017      6.339678   Analog
898  Vinyl Single  Value (Adjusted)  2018      5.386197   Analog
899  Vinyl Single  Value (Adjusted)  2019      6.795946   Analog

[900 rows x 5 columns]

#### Just by dropping the rows with NaN now we went from 2961 rows to 900.

In [18]:
music_sales_df2.to_csv('../data/Usable_Data/Usable_Music_Sales_Data.csv', index=False)

#### This will be my usable data as I can analyze using different Modes and Metrics.